### ISSUE TITLE
Make dockershim periodically clean up its `containerCleanupInfos` map

### ISSUE THREAD

**Issue by wk8** [kubernetes/kubernetes#74843]
**What would you like to be added**:
Periodically remove cleanupInfos for old containers from dockershim's `containerCleanupInfos` map.

One way of doing that would be to just forget about `cleanupInfo`s older than say an hour, as they should always be short-lived.

**Why is this needed**:
https://github.com/kubernetes/kubernetes/pull/74737 introduced a new in-memory map for the dockershim, that could potentially (in pathological cases) cause memory leaks - for containers that use GMSA cred specs, get created successfully, but then never get started nor removed.

This might be okay for the alpha release of the GMSA feature, but should be addressed in the beta.

See https://github.com/kubernetes/kubernetes/pull/74737#discussion_r261784181

**Comment by dims** [comment]
#77886 got moved out. let's move this issue with it as well

/milestone v1.16

**Comment by xmudrii** [comment]
@wk8 Hello! I'm bug triage lead for the 1.16 release cycle and considering this issue is tagged for 1.16, but not updated for a long time, I'd like to check its status. The code freeze is starting on August 29th (about 1.5 weeks from now), which means that there should be a PR ready (and merged) until then.

Do we still target this issue to be fixed for 1.16?

**Comment by wk8** [comment]
@xmudrii : yup we'd love to have this in 1.16, still waiting for kind reviewers on https://github.com/kubernetes/kubernetes/pull/80320 :) ...
